core/client/src/main/java/org/phenotips/remote/client/internal/RemoteMatchFinder.java

```diff
@@ -78,6 +92,45 @@
     @Inject
     private ConsentManager consentManager;
 
+    @Inject
+    private Provider<XWikiContext> provider;
+
+    private XWikiDocument prefsDoc;
+
+    private List<String> remoteIdsList;
+
+    @Override
+    public void initialize() throws InitializationException
+    {
+        this.remoteIdsList = this.getRemotesList();
```

Comment:
This is a bug, since if a server is added while server is running the list of ids will not be updated - the list should be obtained each time the run is started and not cached